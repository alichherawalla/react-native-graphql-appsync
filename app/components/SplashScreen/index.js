import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'
import AppContainer from 'app/components/Container'
import { images } from 'app/themes'
import NavigationService from 'app/services/NavigationService'

const Container = styled(AppContainer)`
  display: flex;
  justify-content: center;
  align-items: center;
`

const Logo = styled.Image`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 200px;
  width: 200px;
`

const SplashScreen = ({ navigation }) => {
  useEffect(() => {
    NavigationService.setTopLevelNavigator(navigation)
  }, [])

  return (
    <Container testID="splash-screen">
      <Logo source={images.velotioLogo} resizeMode="contain" />
    </Container>
  )
}

SplashScreen.propTypes = {
  navigation: PropTypes.object
}
export default SplashScreen
