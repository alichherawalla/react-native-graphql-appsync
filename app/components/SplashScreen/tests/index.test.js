/**
 *
 * Tests for SplashScreen
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import 'jest-dom/extend-expect' // add some helpful assertions

import SplashScreen from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('<SplashScreen />', () => {
  it('Should render and match the snapshot', () => {
    const component = shallow(<SplashScreen />)
    expect(component).toMatchSnapshot()
  })
})
