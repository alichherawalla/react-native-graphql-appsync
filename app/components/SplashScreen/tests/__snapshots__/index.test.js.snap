// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashScreen /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;
